include/albatross/src/core/distribution.hpp

```diff
@@ -99,8 +99,17 @@ struct MarginalDistribution : public DistributionBase<MarginalDistribution> {
     return covariance.diagonal()[i];
   }
 
+  bool approximately_equal(const MarginalDistribution &other,
+                           const double epsilon = 1e-3) const {
+    const bool mean_approx_equal = mean.isApprox(other.mean, epsilon);
+    const bool cov_approx_equal =
+        covariance.diagonal().isApprox(other.covariance.diagonal(), epsilon);
+    return mean_approx_equal && cov_approx_equal;
+  }
+
   bool operator==(const MarginalDistribution &other) const {
-    return (mean == other.mean && covariance == other.covariance);
+    // Use the smallest possible value to approximate
+    return approximately_equal(other, std::numeric_limits<double>::epsilon());
```

Comment:
I think @RReichert  should weigh in here since I think he has a stronger sense of best programming practices than I do.

On one hand treating two distributions equal if their mean and covariance are the same down to float precision seems reasonable to me. On the other, I think it's generally assumed that equality operators will return true only if the two objects are _exactly_ the same.  In other words, maybe `dist_x == dist_y` should behave the same way `double_x == double_y` does (tiny difference lead to inequality). and a user should be choosing `dist_x.approximately_equal(dist_y)` if they don't care about perfect equality?

Reply:
Hmm reading this again, I think having `==` mean exactly equal makes sense, especially since we have `approximately_equal` for float point equality. 

Reverted